[ffmpeg] [Build error] [configure] [test_ld failed]

**Host Environment**
- OS: Windows-10
- Compiler: MSVC-19.0.24215.1
- vcpkg-tool version: 2022-05-05-67e17c1782801cf481be9ac0b3765dff3e4bdeb8
- vcpkg-scripts version: 162a88fa4 2022-07-28
- ffmpeg verson: 4.4.1#14

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install ffmpeg:x86-windows-static`

**Failure logs**
- Console
    ```
    CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
        Command failed: D:/App/vcpkg/downloads/tools/msys2/79d5b6b790550399/usr/bin/bash.exe ./build.sh
        Working Directory: D:/App/vcpkg/buildtrees/ffmpeg/x64-windows-static-rel
        Error code: 1
        See logs for more information:
        D:\App\vcpkg\buildtrees\ffmpeg\build-x64-windows-static-rel-out.log

    Call Stack (most recent call first):
    ports/ffmpeg/portfile.cmake:548 (vcpkg_execute_required_process)
    scripts/ports.cmake:147 (include)
    ```
- build-x64-windows-static-rel-out.log
    ```
    === CONFIGURING ===
    cl.exe is unable to create an executable file.
    C compiler test failed.

    If you think configure made a mistake, make sure you are using the latest
    version from Git.  If the latest version fails, report the problem to the
    ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
    Include the log file "ffbuild/config.log" produced by configure as this will help
    solve the problem.
    ```
- config.log
    ```
    test_ld cc
    test_cc
    BEGIN /c/Users/XXXX/AppData/Local/Temp/ffconf.xmA29ylu/test.c
        1	int main(void){ return 0; }
    END /c/Users//AppData/Local/Temp/ffconf.xmA29ylu/test.c
    cl.exe @D:/App/vcpkg/buildtrees/ffmpeg/x64-windows-static-rel/cflags.rsp -nologo -DHAVE_UNISTD_H=0 -c -Fo/c/Users/XXXX/AppData/Local/Temp/ffconf.xmA29ylu/test.o /c/Users/XXXX/AppData/Local/Temp/ffconf.xmA29ylu/test.c
    test.c
    C:\Users\XXXX\AppData\Local\Temp\ffconf.xmA29ylu\test.c : fatal error C1083: Cannot open compiler generated file: 'D:\c\Users\XXXX\AppData\Local\Temp\ffconf.xmA29ylu\test.o': No such file or directory
    C compiler test failed.
    ```
**Additional context**

The issue is clear that test_ld uses a wrong path[D:\c\Users\XXXX\AppData\Local\Temp\ffconf.xmA29ylu\test.o], which actully is path[C:\Users\XXXX\AppData\Local\Temp\ffconf.xmA29ylu\test.o]

--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -461,7 +461,7 @@ endif()
 set(OPTIONS_CROSS " --enable-cross-compile")
 
 # ffmpeg needs --cross-prefix option to use appropriate tools for cross-compiling.
-if(VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "(/.+)gcc$")
+if(VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "([^\/]*-)gcc$")
     string(APPEND OPTIONS_CROSS " --cross-prefix=${CMAKE_MATCH_1}")
 endif()
 

--- a/ports/ffmpeg/vcpkg.json
+++ b/ports/ffmpeg/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ffmpeg",
   "version": "4.4.1",
-  "port-version": 14,
+  "port-version": 15,
   "description": [
     "a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.",
     "FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created. It supports the most obscure ancient formats up to the cutting edge. No matter if they were designed by some standards committee, the community or a corporation. It is also highly portable: FFmpeg compiles, runs, and passes our testing infrastructure FATE across Linux, Mac OS X, Microsoft Windows, the BSDs, Solaris, etc. under a wide variety of build environments, machine architectures, and configurations."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2250,7 +2250,7 @@
     },
     "ffmpeg": {
       "baseline": "4.4.1",
-      "port-version": 14
+      "port-version": 15
     },
     "ffnvcodec": {
       "baseline": "11.1.5.0",

--- a/versions/f-/ffmpeg.json
+++ b/versions/f-/ffmpeg.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d42747aed3213bf6d62a4a986fb915bdfd854a08",
+      "version": "4.4.1",
+      "port-version": 15
+    },
+    {
       "git-tree": "bd232549fb2bcffed0dcfac1e7e6a54f5a91b5cc",
       "version": "4.4.1",
       "port-version": 14

